### DISCUSSION TITLE
不用 React 组件如何实现点击列头排序？

### DISCUSSION THREAD

**Discussion by nightire** [antvis/S2#1280] [Q&A] [answered]
翻遍了官网上的示例，只有用了 React 组件的例子才有列头上的下拉菜单可以执行排序等操作，因为对 React 不熟所以翻源码找不出具体的实现部分，请教是否有用原生类实现此类需求的例子以供参考？不胜感激！

**Comment by lijinke666** [comment, marked as answer]
原生类提供了 `BaseTooltip`, 你可以继承它实现你自己的 下拉菜单, 然后点击调用 `s2` 提供的排序方法即可


- Tooltip 参考: 

> 可以自定义类, 也可以自定义 content 

 https://s2.antv.vision/zh/docs/manual/basic/tooltip#%E8%87%AA%E5%AE%9A%E4%B9%89

 https://github.com/antvis/S2/blob/814d8aa5c6198c94944fc397825a4473e9cf2bd0/packages/s2-react/src/components/tooltip/custom-tooltip.tsx

  https://github.com/antvis/S2/blob/814d8aa5c6198c94944fc397825a4473e9cf2bd0/packages/s2-react/src/common/constant/options.ts

  https://s2.antv.vision/zh/examples/react-component/tooltip#custom-tooltip

- 排序:

调用 `s2.groupSortByMethod(sortMethod, meta)`

- sortMethod

```ts
export type SortMethod = 'ASC' | 'DESC' | 'asc' | 'desc';
```

- meta

当前单元格信息: https://s2.antv.vision/zh/docs/manual/advanced/get-cell-data#%E8%8E%B7%E5%8F%96%E8%A1%8C%E5%88%97%E6%95%B0%E6%8D%AE

**Reply by nightire** [reply]
谢谢，我没想到 Tooltip 竟然是干这个事情的。以前用过几个这种类型的控件，基本上都是用 Panel/Menu 来代表列头的功能菜单，而 Tooltip 则是 hover 到单元格上时随鼠标显示的浮动窗口，所以看文档硬是没把 Tooltip 和这件事联系起来😂，待我去试试吧。
